code/game/machinery/doors/airlock.dm

```diff
@@ -739,6 +739,8 @@ GLOBAL_LIST_EMPTY(airlock_emissive_underlays)
 			attack_ai(user)
 
 /obj/machinery/door/airlock/CanPass(atom/movable/mover, turf/target, height=0)
+	if(ismob(mover) && mover.checkpass(PASSDOOR) && !locked) // We really dont want mice/drones to get shocked on a door they're passing through
```

Comment:
does it have to be a mob? Since the `pass_flags` already limit what can go through or not

Reply:
Ah I swear I changed that back to istype, I had the same idea